### DISCUSSION TITLE
How Blazor handles the events? `onclick:stopPropagation` cannot prevent native JS events from bubbling up.

### DISCUSSION THREAD

**Discussion by zxyao145** [dotnet/aspnetcore#52956] [Q&A] [answered]
I have two components: the **child** component uses `@onclick` to add event, while the **parent** component uses native JS to add events through IJSRuntime. 

However, using `@onclick:stopPropagation="@true"` on the child component does not prevent the parent component from responding to click events.

Is Blazor's handling of events similar to React?

Child.razor
```razor
<span style="border: 1px solid red;"
      @onclick="OnClick"
      @onclick:stopPropagation="@true">
    child
</span>

@code {
    private void OnClick()
    {
        Console.WriteLine("child click");
    }
}
```

Parent.razor

```razor
@inject IJSRuntime JsRuntime

<div @ref="@_ref" id="parent">
    <Child />
</div>

@code
{

    private ElementReference _ref;
    protected override Task OnAfterRenderAsync(bool firstRender)
    {
        if (firstRender)
        {
            JsRuntime.InvokeVoidAsync("addClick", _ref);
        }
        return base.OnAfterRenderAsync(firstRender);
    }
}
```
`addClick` method:

```javascript
window.addClick = function (dom) {
    dom.addEventListener("click", (e) => {
        console.log("on click", dom.getAttribute("id"));
    })
}
```

the log:
![image](https://github.com/dotnet/aspnetcore/assets/24973084/9937730d-c416-49cc-a699-a8e8284c6f1f)

**Comment by javiercn** [comment, marked as answer]
Blazor relies on event delegation and will listen to the events at the root, which means events bubble all the way before they get stopped. If you want to stop the event at the element level you need to use JS interop to setup a handler on the element itself.

**Reply by zxyao145** [reply]
Thanks very mush!
